src/sentry/tagstore/snuba/backend.py

```diff
@@ -775,9 +777,9 @@ def get_group_tag_value_iter(self, project_id, group_id, environment_id, key, ca
                 ["max", "timestamp", "last_seen"],
             ],
             orderby="-first_seen",  # Closest thing to pre-existing `-id` order
-            # TODO: This means they can't actually iterate all GroupTagValues.
             limit=1000,
             referrer="tagstore.get_group_tag_value_iter",
+            **kwargs
```

Comment:
What is this for?

Reply:
This allows me to pass in `offset` without breaking wherever it's already implemented (otherwise I'd have to add `offset=0` to wherever it's used)

Reply:
Or add `offset=0` as the default param, which didn't seem right to me

Reply:
The explicit parameter enables better mistake detection as you won't get schema validation errors from snuba farther down the callstack. Spreading kwargs into snuba has become a problematic pattern over time as it has made tracking back what parameters each call is using much harder to track down impeding refactoring.

You should also update the interfaces defined in `tagstore/base.py` as well.

Reply:
@markstory Addressed in one of the latest commits, I wasn't sure if it should be added as a default for the definition of `get_group_tag_value_iter()`, or added as offset=0 to every time it's used

I chose the latter but let me know if I should do the opposite.